junit-platform-commons/src/main/java/org/junit/platform/commons/support/conversion/TypeDescriptor.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015-2025 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.support.conversion;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Parameter;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.util.ReflectionUtils;
+
+/**
+ *
+ *
+ * @since 1.13
+ */
+@API(status = EXPERIMENTAL, since = "1.13")
+public final class TypeDescriptor {
+
+	private final Class<?> type;
+
+	public static TypeDescriptor forClass(Class<?> clazz) {
+		return new TypeDescriptor(clazz);
+	}
+
+	public static TypeDescriptor forInstance(Object instance) {
+		return new TypeDescriptor(instance.getClass());
+	}
+
+	public static TypeDescriptor forField(Field field) {
+		return new TypeDescriptor(field.getType());
+	}
+
+	public static TypeDescriptor forParameter(Parameter parameter) {
+		return new TypeDescriptor(parameter.getType());
+	}
+
+	private TypeDescriptor(Class<?> type) {
+		this.type = type;
+	}
+
+	public Class<?> getType() {
+		return type;
+	}
+
+	public Class<?> getWrapperType() {
+		Class<?> wrapperType = ReflectionUtils.getWrapperType(type);
+		return wrapperType != null ? wrapperType : type;
+	}
+
+	public boolean isPrimitive() {
+		return getType().isPrimitive();
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+		TypeDescriptor that = (TypeDescriptor) o;
+		return this.type.equals(that.type);
+	}
+
+	@Override
+	public int hashCode() {
+		return this.type.hashCode();
+	}
```

Comment:
I noticed `EqualsAndHashCodeAssertions` under `junit-jupiter-api` to test `equals`/`hashCode`. 

Would you like me to do something similar in `junit-platform-commons`?

Was [EqualsVerifier](https://jqno.nl/equalsverifier/) ever considered for such tests?

Reply:
> I noticed `EqualsAndHashCodeAssertions` under `junit-jupiter-api` to test `equals`/`hashCode`. It seems I can use the same in `junit-platform-commons`.

:+1: 

> Out of curiosity, was [EqualsVerifier](https://jqno.nl/equalsverifier/) ever considered for such tests?

I don't think it was discussed but it might be a good addition or even replacement but probably in a separate PR.